### PR TITLE
Build: Make QtWebKit the default package to look for and add build switch to…

### PR DESCRIPTION
**Pull request by mlampert**
… force QtWebEngine.

see https://forum.freecadweb.org/viewtopic.php?f=4&t=35386#p298607

### CODE REVIEW

**Comment by wwmayer** [comment]
For the automatic check (i.e. none of the option are activated) the REQUIRED keyword of find_package(Qt5WebKitWidgets REQUIRED) must be dropped because if not found CMake will stop with an error.

**Comment by wwmayer** [comment]
There is also a typo in FREECAD_USE_QWEBKENGINE so that it's not possible any more to activate QtWebEngine.
I have reworked your PR to fix the issues and implemented a combo box to make it easier and more intuitive to choose between the modules.
See 45a5bb6a1

**Comment by wwmayer** [comment]
Merged.

**Comment by mlampert** [comment]
Thanks a lot for the cleanup. I thought I had tested all combinations - sorry about that.

File: CMakeLists.txt [hunk 1]
```diff
@@ -893,13 +894,18 @@ endif()
...
                     find_package(Qt5WebKitWidgets REQUIRED)
-                else()
+                elseif(FREECAD_USE_QWEBENGINE)
                     find_package(Qt5WebEngineWidgets REQUIRED)
                     if (Qt5WebEngineWidgets_VERSION VERSION_LESS 5.7.0)
-                        message(WARNING "** Minimum supported Qt5WebEngine version is 5.70!\n"
-                                "-- Trying with QWebkit, force with FREECAD_USE_QWEBKIT=On")
-
-                        find_package(Qt5WebKitWidgets REQUIRED)
+                        message(FATAL_ERROR "** Minimum supported Qt5WebEngine version is 5.70!\n")
```
Comment (luzpaz) [inline CMakeLists.txt]: Shouldn't this be `5.7.0` ?

File: CMakeLists.txt [hunk 2]
```diff
@@ -893,13 +894,18 @@ endif()
...
+                    endif()
+                else()
+                    find_package(Qt5WebKitWidgets REQUIRED)
+                    if(NOT Qt5WebKitWidgets_FOUND)
+                        find_package(Qt5WebEngineWidgets REQUIRED)
+                        if (Qt5WebEngineWidgets_VERSION VERSION_LESS 5.7.0)
+                            message(FATAL_ERROR "** Minimum supported Qt5WebEngine version is 5.70!\n")
```
Comment (luzpaz) [inline CMakeLists.txt]: Shouldn't this be `5.7.0` ?

**Pull request closed without merging**